src/state/nftMarket/helpers.ts

```diff
@@ -254,101 +351,29 @@ export const fetchWalletTokenIdsForCollections = async (
   return walletNfts.flat()
 }
 
-/**
- * Get the most recently listed NFTs
- * @param first Number of nfts to retrieve
- * @returns NftTokenSg[]
- */
-export const getLatestListedNfts = async (first: number): Promise<NftTokenSg[]> => {
-  try {
-    const res = await request(
-      GRAPH_API_NFTMARKET,
-      gql`
-        query getLatestNftMarketData($first: Int) {
-          nfts(orderBy: updatedAt , orderDirection: desc, first: $first) {
-            ${getBaseNftFields()}
-            collection {
-              id
-            }
-          }
-        }
-      `,
-      { first },
-    )
-
-    return res.nfts
-  } catch (error) {
-    console.error('Failed to fetch NFTs market data', error)
-    return []
-  }
-}
-
-/**
- * Fetch a NFT using the API
- * /!\ THIS FUNCTION WILL BE REPLACED BY AN ENDPOINT ALLOWING TO FETCH MULTIPLE NFTS /!\
- * @param collectionAddress
- * @param tokenId
- * @returns NFT from API
- */
-const getNftApi = async (collectionAddress: string, tokenId: string) => {
-  const res = await fetch(`${API_NFT}/collections/${collectionAddress}/tokens/${tokenId}`)
-  if (res.ok) {
-    const json = await res.json()
-    return json.data
-  }
-
-  console.error("Can't fetch NFT API", res.status)
-  return {}
-}
-
-/**
- * Fetch a list of NFT from different collections
- * @param from Array of { collectionAddress: string; tokenId: string }
- * @returns Array of NFT from API
- */
-export const getNftsFromDifferentCollectionsApi = async (
-  from: { collectionAddress: string; tokenId: string }[],
-): Promise<NFT[]> => {
-  const promises = from.map((nft) => getNftApi(nft.collectionAddress, nft.tokenId))
-  const responses = await Promise.all(promises)
-  return responses.map((res, index) => ({
-    tokenId: res.tokenId,
-    name: res.name,
-    collectionName: res.collection.name,
-    collectionAddress: from[index].collectionAddress,
-    description: res.description,
-    updatedAt: res.updatedAt,
-    attributes: res.attributes,
-    image: {
-      original: res.image?.original,
-      thumbnail: res.image?.thumbnail,
-    },
-  }))
-}
-
 /**
  * Helper to combine data from the collections' API and subgraph
- * TODO: Type this properly
  */
-export const combineCollectionData = (collectionApiData: any, collectionSgData: any): Record<string, Collection> => {
-  const collectionsMarketObj = collectionSgData.reduce(
+export const combineCollectionData = (
+  collectionApiData: ApiCollection[],
+  collectionSgData: CollectionMarketDataBaseFields[],
+): Record<string, Collection> => {
+  const collectionsMarketObj: Record<string, CollectionMarketDataBaseFields> = collectionSgData.reduce(
     (prev, current) => ({ ...prev, [current.id]: { ...current } }),
     {},
   )
 
   return collectionApiData.reduce((accum, current) => {
     const collectionMarket = collectionsMarketObj[current.address.toLowerCase()]
+    const collection: Collection = {
+      ...current,
+      ...collectionMarket,
+      slug: slugify(current.name),
```

Comment:
Not sure if `slug` field is actually used anywhere now, I think there's just one place. Plugged it here since its the only field missing from internal `Collection` type. Previously slug field was undefined. 🤔

Reply:
You can remove it.

Reply:
Oh yeah, you're right, its not used at all, the slug used in that one place was from different `Collection` type.
Removed this one